conda_forge_tick/update_recipe/v2/context.py

```diff
@@ -0,0 +1,19 @@
+import jinja2
+
+
+def load_recipe_context(
+    context: dict[str, str], jinja_env: jinja2.Environment
+) -> dict[str, str]:
+    """
+    Load all string values from the context dictionary as Jinja2 templates.
+    Use linux-64 as default target_platform, build_platform, and mpi.
```

Comment:
I'm concerned that assuming linux as the default will result in invalid updates for some recipes.

Reply:
I understand where you are coming from but we do ignore the selectors in the version updating code (e.g. we go down both branches, `then` and `else`.

However, we do use the value for templating, so people could theoretically do something like 

```yaml
source:
  url: https://foo.${{ "linux" if linux else "bla" }}/...
  sha256: ${{ "foo" ...
```

But that doesn't seem very likely and also buggy so I don't think we'll see that a lot in the wild.

Reply:
I believe that doesn't work in the context section. However you _can_ use templating and inline Jinja expressions, such as:

```
context:
  version: "1.2.3"
  version_string: "v${{ version if linux else "foo" }}"
```

Reply:
right so we need to make this correct.

Reply:
I think it's bad practice in many ways. 

Can I instead add a lint on the sha256 hash being in the proper spot (NOT in context)?
Then people would not / could not do many shenanigans with the URL either.

Reply:
That won't change other items that might be platform dependent in the context section.

Reply:
But the SHA hash will make it impossible, thankfully :)

Reply:
Nope. Folks could define parts of urls in the context section that are platform dependent that are not hashes. So when the bot comes through to render those urls and parts, it will compute the wrong hash since it might have a valid, but incorrect url for that platform.

Reply:
@beckermr - i sorta get where you come from but is it really necessary to care about all of these edge cases on day one?

It is also not a very clever way to write the recipe. You would have to "double-if" the source.

E.g.

```yaml
context:
  foo: ${{ "bla" if linux else "foo" }}

source:
  - if: linux
    then:
      url: ${{ foo }}
      sha256: 1abcd
    else:
      url: ${{ foo }}
      sha256: 2bcde
```

So I don't think many people would wanna write their sources like this (doesn't make much logical sense).

Reply:
Yes we should care. My suggested procedure for this work was to abstract the current version update algorithm to be independent of the recipe format backend and then use that. It should handle most, if not all, of the cases I am bringing up.